nri-redis/src/Redis/Internal.hs

```diff
@@ -354,3 +417,21 @@ traceQuery commands host port task =
                   )
             )
         )
+
+-- | Use SCAN command to find keys matching a pattern, and "fold" over them in batches, producing a result value.
+-- keyMatchPattern       A glob-like pattern to match keys, see https://redis.io/commands/keys/
+-- approxCountPerBatch   A hint for the batch size you want to process at once. Only approximate.
+-- processKeyBatch       Function to process one batch of keys (provided as plain Text without namespace prefix)
+-- initAccumulator       Initial value for the fold accumulator
+foldWithScan :: Handler' x -> Maybe Text -> Maybe Int -> ([Text] -> a -> Task Error a) -> a -> Task Error a
+foldWithScan handler keyMatchPattern approxCountPerBatch processKeyBatch initAccumulator =
+  let go accumulator cursor = do
+        (nextCursor, keyBatch) <-
+          Scan cursor keyMatchPattern approxCountPerBatch
+            |> query handler
+        nextAccumulator <-
+          processKeyBatch keyBatch accumulator
+        if nextCursor == Database.Redis.cursor0
+          then Task.succeed nextAccumulator
+          else go nextAccumulator nextCursor
+   in go initAccumulator Database.Redis.cursor0
```

Comment:
I'm not sure how well `foldWithScan` fits with the overall API.

I experimented with creating a monad instance instead, but got stuck implementing `cmds` and `mapKeys`. Not sure how to do those without actually running the queries.

In my first attempt I tried creating an `AndThen` constructor. Got stuck, tried with a `Join` constructor instead, got stuck again. Along the way, I realised that `Monad` creates a lot of ways of composing things, and all of that needs to be tested. It simplifies matters quite a bit to just directly support the exact thing I need.

Also, running it in Task allows you to do anything you like with the keys, not just other Redis queries.